modules/geochemistry/include/utils/EquilibriumConstantInterpolator.h

```diff
@@ -0,0 +1,64 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+#include "LeastSquaresFitBase.h"
+#include "DualRealOps.h"
+
+/**
+ * Fit the equilibrium constant values read from the thermodynamic databse at specified
+ * temperature values with one of two types of fits:
+ *
+ * a fourth-order polynomial fit (for GWB databses)
+ *
+ * log(K)= a_0 + a_1 T + a_2 T^2 + a_3 T^3 + a_4 T^4
+ *
+ * a Maier-Kelly type fit (for EQ3/6 databases)
+ *
+ * log(K)= a_0 ln(T) + a_1 + a_2 T + a_3 / T + a_4 / T^2
```

Comment:
Just check that it's "ln" and not "log10" - geochemists love log10 !

Reply:
I think its natural log but will double check!